### DISCUSSION TITLE
Improve dry-run to better support local conf development

### DISCUSSION THREAD

**Discussion by JanoschDeurer** [renovatebot/renovate#27333] [Suggest an Idea]
### Tell us more.

As per the documentation the `--platform=local` option is for developing and testing renovate configuration files locally. This is great for finding obvious errors: Invalid json and other forms of invalid configuration. However for more advanced problems the output provides little help. I think it would be tremendously helpful to have the following data for each update that would happen:

- Name of the dependecy (depName)
- What is the current version (currentValue)
- What is the version after the update
- What is the datasource for the update
- What is the update type (major, minor, pin ect.)
- Is there a schedule defined
- What type of versioning is used
- Is a merge request created
  - Will it be automerged

I know that some of this information is already available when running in debug mode. However, in debug mode it is lost between all the other debug messages. As rewriting of the config is the most common thing for most when working with renovate, I think it would be great to have a tailored output for this use case. It could be either implemented as additional info level log output for the local platform or with a separate command line option.

**Comment by rarkins** [comment]
Most of this is in the "packageFiles with updates" log message and we don't plan to duplicate that

**Reply by JanoschDeurer** [reply]
Thanks for the fast response. I get, that you don't want to duplicate any log messages. With a separate command line option you could support the use case without duplicating the log message. Additionally the information about schedules and merge requests that I wrote about is missing altogether. It is not even available in the debug log.

I would be really happy if we found a way to support that use case.

**Reply by rarkins** [reply]
If you can focus on the info that's missing, we can look into those, but right now it's mixed up with information which is already available

**Reply by JanoschDeurer** [reply]
Ok, lets do this, should I close this and open a new discussion that is limited to the missing information?

**Reply by rarkins** [reply]
It's fine to continue here or open new - whatever you prefer in this case
